workers/unity/Assets/Gdk/Core/GameObjectRepresentation/EntityGameObjectLifecycleSystem.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    public class EntityGameObjectLifecycleSystem : ComponentSystem
+    {
+        private uint currentHandle;
+        private readonly Dictionary<uint, GameObject> entityGameObjectCache = new Dictionary<uint, GameObject>();
+
+        public struct AddedEntitiesData
+        {
+            public int Length;
+            public EntityArray Entities;
+            [ReadOnly] public ComponentArray<GameObjectReference> GameObjectReferences;
```

Comment:
This should be added at the same time as the handle.

Reply:
That would mean that either the `entityGameObjectCache` needs to be written to by from multiple places which doesn't separate concerns nicely or that `EntityGameObjectManager.LinkGameObjectToEntity()` is provided by the `EntityGameObjectLifecycleSystem` in which case calling that method requires you to have a reference to a system. That also doesn't sound very idiomatic. Is ther something else that you have in mind?

Reply:
DIscussed offline. Moved gameobject deletion into the GameObjectInitializationSystem.